src/secondary.rs

```diff
@@ -689,6 +689,43 @@ impl<K: Key, V> SecondaryMap<K, V> {
             inner: self.iter_mut(),
         }
     }
+
+    /// An entry API to access an element of the map. Returns None if the key is invalid.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use slotmap::*;
+    /// # use std::collections::HashSet;
+    /// let mut sm = SlotMap::new();
+    /// let mut sec = SecondaryMap::new();
+    /// let k = sm.insert(1);
+    /// let v = sec.entry(k).unwrap().or_insert(10);
+    /// assert_eq!(*v, 10);
+    /// ```
+    pub fn entry(&mut self, key: K) -> Option<Entry<K, V>> {
+        let keyd = key.clone().into();
+
+        if let Some(slot) = self.slots.get_mut(keyd.idx as usize) {
+            if is_older_version(keyd.version.get(), slot.version) {
+                return None;
+            }
+        }
+
+        Some(if self.contains_key(key.clone()) {
```

Comment:
Can't this `contains_key` lookup be avoided by merging into the previous section?